frame/support/src/storage/mod.rs

```diff
@@ -303,6 +303,17 @@ pub trait IterableStorageMap<K: FullEncode, V: FullCodec>: StorageMap<K, V> {
 	///
 	/// NOTE: If a value fail to decode because storage is corrupted then it is skipped.
 	fn translate<O: Decode, F: FnMut(K, O) -> Option<V>>(f: F);
+
+	/// Translate the next entry following `previous_key` by a function `f`.
+	/// By returning `None` from `f` for an element, you'll remove it from the map.
+	///
+	/// Returns the next key to iterate from in lexicographical order of the encoded key.
+	/// and whether or not the translation failed because either the old key or value was
```

Comment:
Sound rather like `try_translate_next`.

Reply:
Yeah or not return the `Result`. As the user is in control of `f` they can also find out if it "failed".